ddtrace/bootstrap/sitecustomize.py

```diff
@@ -111,7 +188,22 @@ def update_patched_modules():
         update_patched_modules()
         from ddtrace import patch_all
 
+        # We need to clean up after we have imported everything we need from
+        # ddtrace, but before we register the patch-on-import hooks for the
+        # integrations. This is because registering a hook for a module
+        # that is already imported causes the module to be patched immediately.
+        # So if we unload the module after registering hooks, we effectively
+        # remove the patching, thus breaking the tracer integration.
+        if will_run_module_cloning:
+            cleanup_loaded_modules()
+
         patch_all(**EXTRA_PATCHED_MODULES)
+    else:
+        if will_run_module_cloning:
```

Comment:
can we just add this to L186 right before the `if_trace_enabled:` call?

we already do `from ddtrace import tracer` above, so we don't need to really guard around the import of `ddtrace.patch_all`.


I think ?

Reply:
`update_patched_modules` isn't importing anything either.